packages/ERTP/util/store.js

```diff
@@ -0,0 +1,41 @@
+// Copyright (C) 2019 Agoric, under Apache license 2.0
+
+import harden from '@agoric/harden';
+
+import { insist } from './insist';
+/**
+ * Distinguishes between adding a new key (init) and updating or
+ * referencing a key (get, set, delete).
+ *
+ * `init` is only allowed if the key does not already exist. `Get`,
+ * `set` and `delete` are only allowed if the key does already exist.
+ * @param  {string} keyName - the column name for the key
+ */
+function makeStore(keyName = 'key') {
+  const wm = new WeakMap();
+  const insistKeyDoesNotExist = key =>
+    insist(!wm.has(key))`${keyName} already registered`;
+  const insistKeyExists = key =>
+    insist(wm.has(key))`${keyName} not found: ${key}`;
```

Comment:
I agree that `keyName` should be included in the thrown error. Is this an issue only for `insistKeyDoesNotExist` and `insistKeyExists`? If so, then you can do something a bit ugly but self contained for those two functions. 

```js
  const insistKeyDoesNotExist = key =>
    insist(!wm.has(key))([`${keyName} already registered`]);
  const insistKeyExists = key =>
    insist(wm.has(key))([`${keyName} not found: `, ''], key);
```

This just expands the tagged template syntax into the function call it is approximately equivalent to, including the `keyName` into the literal template part of the template expression rather than as a substitution value. I leave `key` as a substitution value since it may not be safe to generally reveal. Please test to see if it does what you described.

Reply:
great! I will use this